dace/dtypes.py

```diff
@@ -260,159 +260,43 @@ def __repr__(self):
-    typeclass(numpy.float32),
-    frozenset((typeclass(numpy.float32), typeclass(numpy.float16))):
-    typeclass(numpy.float32),
-    frozenset((typeclass(numpy.float16), typeclass(numpy.float16))):
-    typeclass(numpy.float16),
-    frozenset((typeclass(numpy.int8), typeclass(numpy.float16))):
-    typeclass(numpy.float16),
-    frozenset((typeclass(numpy.int16), typeclass(numpy.float16))):
-    typeclass(numpy.float16),
-    frozenset((typeclass(numpy.int32), typeclass(numpy.float16))):
-    typeclass(numpy.float16),
-    frozenset((typeclass(numpy.uint8), typeclass(numpy.float16))):
-    typeclass(numpy.float16),
-    frozenset((typeclass(numpy.uint16), typeclass(numpy.float16))):
-    typeclass(numpy.float16),
-    frozenset((typeclass(numpy.uint32), typeclass(numpy.float16))):
-    typeclass(numpy.float16),
-    frozenset((typeclass(numpy.int8), typeclass(numpy.float32))):
-    typeclass(numpy.float32),
-    frozenset((typeclass(numpy.int16), typeclass(numpy.float32))):
-    typeclass(numpy.float32),
-    frozenset((typeclass(numpy.int32), typeclass(numpy.float32))):
-    typeclass(numpy.float32),
-    frozenset((typeclass(numpy.uint8), typeclass(numpy.float32))):
-    typeclass(numpy.float32),
-    frozenset((typeclass(numpy.uint16), typeclass(numpy.float32))):
-    typeclass(numpy.float32),
-    frozenset((typeclass(numpy.uint32), typeclass(numpy.float32))):
-    typeclass(numpy.float32),
-    frozenset((typeclass(numpy.int8), typeclass(numpy.float64))):
-    typeclass(numpy.float64),
-    frozenset((typeclass(numpy.int16), typeclass(numpy.float64))):
-    typeclass(numpy.float64),
-    frozenset((typeclass(numpy.int32), typeclass(numpy.float64))):
-    typeclass(numpy.float64),
-    frozenset((typeclass(numpy.uint8), typeclass(numpy.float64))):
-    typeclass(numpy.float64),
-    frozenset((typeclass(numpy.uint16), typeclass(numpy.float64))):
-    typeclass(numpy.float64),
-    frozenset((typeclass(numpy.uint32), typeclass(numpy.float64))):
-    typeclass(numpy.float64),
-}
+def result_type_of(lhs, rhs):
+    """Returns the largest between two types (dace.types.typeclass) according
+       to C semantics."""
+    if lhs == rhs:
+        return lhs  # Types are the same, return either
+    if lhs == None:
+        return rhs  # Use RHS even if it's None
+    if rhs == None:
+        return lhs  # Use LHS
+    # Extract the numpy type so we can call issubdtype on them
+    lhs_ = lhs.type if isinstance(lhs, typeclass) else lhs
+    rhs_ = rhs.type if isinstance(rhs, typeclass) else rhs
+    # Both are integers
+    if numpy.issubdtype(lhs_, numpy.integer) and numpy.issubdtype(
+            rhs_, numpy.integer):
+        # If one byte width is larger, use it
+        if lhs_.itemsize > rhs_.itemsize:
+            return lhs
+        elif lhs_.itemsize < rhs_.itemsize:
+            return rhs
+        # Sizes are the same
+        if numpy.issubdtype(lhs_, numpy.unsignedinteger):
```

Comment:
what about the other side of this?

Reply:
What do you mean?

Reply:
if `rhs_` is unsigned but `lhs_` isn't.

Reply:
Then we have to return `rhs_`, which is what we do

Reply:
It's in the comments :-)